core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -46,7 +47,9 @@
   private final SupportsNamespaces nsDelegate;
 
   public RESTCatalog() {
-    this(SessionCatalog.SessionContext.createEmpty(), new HTTPClientFactory());
+    this(
+        SessionCatalog.SessionContext.createEmpty(),
+        config -> HTTPClient.builder().uri(config.get(CatalogProperties.URI)).build());
```

Comment:
Do we need to call `uri` or can that be pulled out of the config by the builder?

Reply:
Are you suggesting just passing the config to the builder instead of passing the URI?  Currently, that's the only value that would be used.

Reply:
You're right. Nevermind. I thought we were passing both for some reason.